### PR TITLE
Added addon for creating default StorageClass depending on the cloud …

### LINKED ISSUE #1666: Installing the correct default StorageClass into the cluster via an extra addon
Depending on which platform we're deploying a user cluster, we need the correct pre-installed default storage class. 
The StorageClass should be installed via an extra addon, that's using the Go templating engine to render the StorageClass manifest.

### PR DESCRIPTION
**Pull request by glower**
…provider

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1666

**Special notes for your reviewer**:

**Release note**:

```release-note
the correct default StorageClass is now installed into the user cluster via an extra addon
```

### CODE REVIEW

**Comment by glower** [comment]
Found here: https://kubernetes.io/docs/concepts/storage/storage-classes/ almost everything except Hetzner.

After merge need to release addons and adapt docu

File: addons/default-storage-class/starage-class.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    type: gp2 # io1, gp2, sc1, st1 
+    zone: {{ .Cluster.Spec.Cloud.AWS.AvailabilityZone }}
+{{ end }}
+{{ if .Cluster.Spec.Cloud.VSphere }}
+  provisioner: kubernetes.io/vsphere-volume
+  parameters:
+    diskformat: thin
+{{ end }}
+{{ if .Cluster.Spec.Cloud.Openstack }}
+  provisioner: kubernetes.io/cinder
```
Comment (mrIncompetent) [inline addons/default-storage-class/starage-class.yaml]: The type does not need to be specified?

Comment (glower) [reply addons/default-storage-class/starage-class.yaml]: according docu:
```
type: VolumeType created in Cinder. Default is empty
```
And I did not find a list of valid `VolumeType`s

File: addons/default-storage-class/storage-class.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+{{ if .Cluster.Spec.Cloud.VSphere }}
+  provisioner: kubernetes.io/vsphere-volume
+  parameters:
+    diskformat: thin
+{{ end }}
+{{ if .Cluster.Spec.Cloud.Openstack }}
+  provisioner: kubernetes.io/cinder
+{{ end }}
+{{ if .Cluster.Spec.Cloud.Digitalocean }}
+  provisioner: kubernetes.io/digitalocean-volume
```
Comment (kron4eg) [inline addons/default-storage-class/storage-class.yaml]: that requires 3rd party controller

Comment (mrIncompetent) [reply addons/default-storage-class/storage-class.yaml]: Yes, please remove it

Comment (glower) [reply addons/default-storage-class/storage-class.yaml]: Fixed

**Comment by glower** [comment]
Do I first need to make a release of the addons? Otherwise `./api/hack/verify-addon-version.sh` fails all the time

File: api/pkg/controller/addon/addon_controller_test.go [hunk 3]
```diff
@@ -92,6 +95,26 @@ var (
 	combinedTestManifest = fmt.Sprintf("%s---\n%s\n---\n%s", testManifest1, testManifest2, testManifest3)
 )
 
+func TestDefaultStoragrClass(t *testing.T) {
+	data := &templateData{
+		Cluster: setupTestCluster(""),
+	}
+	filename := "/home/igor/go/src/github.com/kubermatic/kubermatic/addons/default-storage-class/starage-class.yaml"
```
Comment (mrIncompetent) [inline api/pkg/controller/addon/addon_controller_test.go]: ...

File: addons/default-storage-class/storage-class.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  annotations:
+    storageclass.beta.kubernetes.io/is-default-class: "true"
+  labels:
+    kubernetes.io/cluster-service: "true"
+  name: default
+{{ if .Cluster.Spec.Cloud.Azure }}
+  provisioner: kubernetes.io/azure-disk
```
Comment (mrIncompetent) [inline addons/default-storage-class/storage-class.yaml]: I haven't run this code, but according to https://kubernetes.io/docs/concepts/storage/storage-classes/

The keys must be on the root of the object, and must not belong to metadata?

File: addons/default-storage-class/storage-class.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  annotations:
+    storageclass.beta.kubernetes.io/is-default-class: "true"
+  labels:
+    kubernetes.io/cluster-service: "true"
+  name: default
```
Comment (mrIncompetent) [inline addons/default-storage-class/storage-class.yaml]: name must be `standard`

File: addons/default-storage-class/storage-class.yaml (hunk 4 above)
Comment (glower) [reply addons/default-storage-class/storage-class.yaml]: Yes, this is correct, fixed it

File: addons/default-storage-class/storage-class.yaml (hunk 5 above)
Comment (glower) [reply addons/default-storage-class/storage-class.yaml]: fixed

**Comment by glower** [comment]
For testing the new addon was deployed on `run.cubermatic.io` , created user cluster on AWS and vSphere and successfully created PVC's  on this clusters

**Review by mrIncompetent — APPROVED** [review]

**Comment by mrIncompetent** [comment]
Should be cherry picker into the v2.6 release branch

**Review by kron4eg — APPROVED** [review]

**Pull request merged**